iot_config/src/route_service.rs

```diff
@@ -390,49 +383,29 @@ impl iot_config::Route for RouteService {
 
         tracing::info!("client subscribed to route stream");
         let pool = self.pool.clone();
-        let shutdown_listener = self.shutdown.clone();
         let (tx, rx) = tokio::sync::mpsc::channel(20);
         let signing_key = self.signing_key.clone();
 
         let mut route_updates = self.subscribe_to_routes();
 
         tokio::spawn(async move {
-            tokio::select! {
-                _ = shutdown_listener.clone() => return,
-                result = stream_existing_routes(&pool, &signing_key, tx.clone())
-                             .and_then(|_| stream_existing_euis(&pool, &signing_key, tx.clone()))
-                             .and_then(|_| stream_existing_devaddrs(&pool, &signing_key, tx.clone()))
-                             .and_then(|_| stream_existing_skfs(&pool, &signing_key, tx.clone())) => {
-                    if let Err(error) = result {
-                        tracing::error!(?error, "Error occurred streaming current routing configuration");
-                        return;
-                    }
-                }
+            if stream_existing_routes(&pool, &signing_key, tx.clone())
+                .and_then(|_| stream_existing_euis(&pool, &signing_key, tx.clone()))
+                .and_then(|_| stream_existing_devaddrs(&pool, &signing_key, tx.clone()))
+                .and_then(|_| stream_existing_skfs(&pool, &signing_key, tx.clone()))
+                .await
+                .is_err()
```

Comment:
why not just call this combinator chain, map_err at the end where we keep the error trace and then `await?`?